profile/src/main/java/rdx/works/profile/data/model/factorsources/FactorSource.kt

```diff
@@ -8,14 +8,23 @@ import kotlinx.serialization.ExperimentalSerializationApi
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import rdx.works.core.HexCoded32Bytes
+import rdx.works.core.Identified
 import rdx.works.profile.data.model.MnemonicWithPassphrase
 import rdx.works.profile.data.model.compressedPublicKey
 import rdx.works.profile.data.model.pernetwork.DerivationPath
 import rdx.works.profile.data.utils.hashToFactorId
 import java.time.Instant
 
 @Serializable(with = FactorSourceSerializer::class)
-sealed class FactorSource {
+sealed class FactorSource : Identified {
+
+    override val identifier: String
+        get() = when (this) {
+            is DeviceFactorSource -> id.body.value
+            is LedgerHardwareWalletFactorSource -> id.body.value
+            is OffDeviceMnemonicFactorSource -> id.body.value
+            is TrustedContactFactorSource -> id.body.value
+        }
```

Comment:
I think a simple `id.body.value` would suffice since all factor sources implement the identifier, don't they?

Reply:
no, I've added identifier implementation at `FactorSource` level. Do you think it is better to implement `identifier` at each individual factor source?